citar.el

```diff
@@ -834,15 +833,24 @@ With prefix, rebuild the cache before offering candidates."
   (bibtex-completion-insert-bibtex
    (citar--extract-keys keys-entries)))
 
+(defun citar--insert-bibtex (key)
+  "Insert the bibtex entry for KEY at point."
+  ;; TODO: this doesn't work correctly.
+  (with-temp-buffer
+    (citar--open-entry key)
+    (bibtex-copy-entry-as-kill))
+  (bibtex-yank))
+
 ;;;###autoload
 (defun citar-add-pdf-attachment (keys-entries)
   "Attach PDF(s) associated with the KEYS-ENTRIES to email.
 With prefix, rebuild the cache before offering candidates."
   (interactive (list (citar-select-refs
                       :rebuild-cache current-prefix-arg)))
- (bibtex-completion-add-PDF-attachment
-  (citar--extract-keys
-   keys-entries)))
+  (unless (require 'bibtex-completion nil 'noerror)
+    (error "This is a deprecated command that relies on 'bibtex-completion'"))
+  (bibtex-completion-add-PDF-attachment
```

Comment:
I think you also need a `declare-function` statement to avoid bytecode compilation warnings.